File: backend/src/main/java/com/example/evaluacion2/usuarios/entity/UsuarioEntity.java
package com.example.evaluacion2.usuarios.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "usuarios")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class UsuarioEntity {
    @Id
    private String email;
    
    @Column(nullable = false, unique = true)
    private String username;
    
    @Column(nullable = false)
    private String nombre;
    
    @Column(nullable = false)
    private String password;
    
    @Column(nullable = false)
    private Boolean isAdmin = false;
}

File: app/src/main/java/com/example/evaluacion2/data/entities/UsuarioEntity.kt
package com.example.evaluacion2.data.entities

import androidx.room.Entity
import androidx.room.PrimaryKey

import androidx.room.Index

@Entity(
    tableName = "usuarios",
    indices = [Index(value = ["username"], unique = true)]
)
data class UsuarioEntity(
    @PrimaryKey val email: String,
    val username: String,
    val nombre: String,
    val password: String,
    val isAdmin: Boolean = false
)
